neqo-transport/src/send_stream.rs

```diff
@@ -718,7 +750,7 @@ impl SendStream {
                 builder.encode_vvec(&data[..length]);
             }
             debug_assert!(builder.len() <= builder.limit());
-
+            self.bytes_sent += length as u64;
```

Comment:
This will overcount if you have packet loss.

That suggests that you want to add a test case where you need to retransmit lost data, so that you can ensure that this value correctly matches the amount of data that has been sent.

I think that the right answer - if we can't track this some other way - is:

```rust
self.bytes_sent = max(self.bytes_sent, offset + u64::try_from(length).unwrap());
```

Edit:  I just realized that you don't need a variable for this either.  This is already tracked by our flow control.  `fc.used()` should give you the same value (ignore the connection-level flow control and just look at the stream level).  That value is carried over into the final size.

Reply:
It seems that using `fc.used()` is not correct here, because `used` value is increased when [consume](https://searchfox.org/mozilla-central/rev/aec3a901e6f6b3041b5ec457c9111a042cef1fb1/third_party/rust/neqo-transport/src/send_stream.rs#993) is called.
I'll put this line in `mark_as_sent`, since it's clearer and easier to write a test for it.
```
self.bytes_sent = max(self.bytes_sent, offset + u64::try_from(length).unwrap());
```